lib/core.js

```diff
@@ -469,6 +469,7 @@ module.exports = class Core {
 
     const appended = batch.length > batch.ancestors
 
+    this.header.contiguousLength = batch.length
```

Comment:
I think this should be Math.min(batch.ancestors, contigLength)

Reply:
You can test that by checking the contig len on a peer that is replicating if the writable one truncates